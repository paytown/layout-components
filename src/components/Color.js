import React from 'react';

export default function Color() {
  const name = 'sunshine';
  const rgb = {
    red: 255,
    green: 299,
    blue: 0
  };

  const style = {
    backgroundColor: `rgba(${rgb.red}, ${rgb.green}, ${rgb.blue}, 1)`,
    borderRadius: '10px',
    width: '50px',
    height: '50px'
  };

  return (
    <>
      <h2>
        {name}
      </h2>
      <div style={style}></div>
      <h3>RGB</h3>
      <ul>
        <li>Red: {rgb.red}</li>
        <li>Green: {rgb.green}</li>
        <li>Blue: {rgb.blue}</li>
      </ul>
    </>
  );
}
