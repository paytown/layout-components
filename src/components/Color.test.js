import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('color test', () => {
  it('testing for color', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
