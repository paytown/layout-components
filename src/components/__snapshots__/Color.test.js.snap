// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color test testing for color 1`] = `
<Fragment>
  <h2>
    sunshine
  </h2>
  <div
    style={
      Object {
        "backgroundColor": "rgba(255, 299, 0, 1)",
        "borderRadius": "10px",
        "height": "50px",
        "width": "50px",
      }
    }
  />
  <h3>
    RGB
  </h3>
  <ul>
    <li>
      Red: 
      255
    </li>
    <li>
      Green: 
      299
    </li>
    <li>
      Blue: 
      0
    </li>
  </ul>
</Fragment>
`;
